Cargo.toml

```diff
@@ -21,7 +21,7 @@ homepage = "https://github.com/holochain/kitsune2"
 keywords = ["holochain", "kitsune", "p2p", "dht", "networking"]
 categories = ["network-programming"]
 license = "Apache-2.0"
-edition = "2021"
+edition = "2024"
```

Comment:
This is why there are so many changes, it's 90% formatter changes and a few places where lifetimes can now be elided.